### PR TITLE
Dev merge template tools 2.0.1

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](<https://github.com/>nf-core/rnavar/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/rnavar _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Merging template.
Hopefully not destroying most of @praveenraj2018's work

**Comment by maxulysse** [comment]
ok, so pipeline hangs waiting for some channel, I definitively must have broke something.
I'll try fixing that up.

**Comment by praveenraj2018** [comment]
> ok, so pipeline hangs waiting for some channel, I definitively must have broke something.
> I'll try fixing that up.

OK.

**Comment by maxulysse** [comment]
@praveenraj2018 any idea what might caused:
```
A USER ERROR has occurred: Read SRR5665260.1.3389994 chr22:2025-2146 is malformed: The input .bam file contains reads with no platform information. First observed at read with name = SRR5665260.1.3389994
```
cf: https://github.com/nf-core/rnavar/pull/7/checks?check_run_id=3192769586

**Review by praveenraj2018 — COMMENTED** [review]
Overall it is good. Once the `STAR/align` module is updated, we will have to add `params.seq_platform` in the config and run the test again.

**Comment by praveenraj2018** [comment]
you need to add `seq_platform = "illumina"` in the nextflow.config file.

**Comment by maxulysse** [comment]
> you need to add `seq_platform = "illumina"` in the nextflow.config file.

Definitively a better idea than hardcoding it in the workflow.
I'll do that right away

**Comment by maxulysse** [comment]
@praveenraj2018 Made good progress, just one last issue with the last process.
Any idea?

**Comment by praveenraj2018** [comment]
I will check.

File: nextflow.config [hunk 1]
```diff
@@ -75,8 +42,8 @@ params {
     //GATK variant filter parameters
     window                     = 35
     cluster                    = 3
-    fs_filter                  = 30.0
-    qd_filter                  = 2.0
+    fs_filter                  = 30
```
Comment (praveenraj2018) [inline nextflow.config:46]: fs_filter = 30.0

File: nextflow_schema.json [hunk 2]
```diff
@@ -736,34 +276,139 @@
...
+        "window": {
+            "type": "integer",
+            "default": 35
+        },
+        "cluster": {
+            "type": "integer",
+            "default": 3
+        },
+        "fs_filter": {
+            "type": "integer",
```
Comment (praveenraj2018) [inline nextflow_schema.json:354]: "type": "number",
"default": 30.0

File: nextflow_schema.json [hunk 3]
```diff
@@ -736,34 +276,139 @@
...
+            "default": 30
+        },
+        "qd_filter": {
```
Comment (praveenraj2018) [inline nextflow_schema.json:357]: "type": "number",
"default": 2.0

**Review by praveenraj2018 — COMMENTED** [review]
Error is due to the number format in filter values of QD and FS. It needs to be float 30.0 and 2.0 for FS and QD respectively. Changes highlighted. Type has to be changed in schema json file also.

**Review by praveenraj2018 — APPROVED** [review]
I think this is good draft.

**Pull request merged**